types/oidcsettings.pp

```diff
@@ -1,113 +1,148 @@
+    Optional['ClaimPrefix']                                => String,
+    Optional['ClaimDelimiter']                             => Pattern[/^.$/],
+    Optional['RemoteUserClaim']                            => String,
+    Optional['PassIDTokenAs']                              => Pattern[/^((claims|payload|serialized)\s?)+$/],
+    Optional['PassUserInfoAs']                             => Pattern[/^((claims|json(:([A-Za-z0-9\-\._])+)?|(signed_)?jwt(:([A-Za-z0-9\-\._])+)?)\s?)+$/],
+    Optional['PassClaimsAs']                               => Enum['none', 'headers', 'environment', 'both'],
+    Optional['AuthNHeader']                                => String,
+    Optional['HTTPTimeoutLong']                            => Integer,
+    Optional['HTTPTimeoutShort']                           => Integer,
+    Optional['StateTimeout']                               => Integer,
+    Optional['ScrubRequestHeaders']                        => Enum['On', 'Off'],
+    Optional['OutgoingProxy']                              => String,
+    Optional['UnAuthAction']                               => Pattern[/^(auth|pass|401|407|410)\s.*/],
+    Optional['UnAutzAction']                               => Pattern[/^(none|headers|environment|both)(\s+(latin1|base64url|none)+)?$/],
+    Optional['PreservePost']                               => Enum['On', 'Off'],
+    Optional['PreservePostTemplates']                      => String,
+    Optional['PassRefreshToken']                           => Enum['On', 'Off'],
+    Optional['RequestObject']                              => String,
+    Optional['ProviderMetadataRefreshInterval']            => Integer,
+    Optional['InfoHook']                                   => Pattern[/^((iat|access_token|access_token_expires|id_token|id_token_hint|userinfo|refresh_token|exp|timeout|remote_user|session)\s?)+$/],
+    Optional['BlackListedClaims']                          => String,
+    Optional['WhiteListedClaims']                          => String,
+    Optional['RefreshAccessTokenBeforeExpiry']             => Pattern[/^[0-9]+(\s(logout_on_error|authenticate_on_error|502_on_error))?$/],
+    Optional['XForwardedHeaders']                          => String,
+    Optional['CABundlePath']                               => String,
+    Optional['DefaultLoggedOutURL']                        => String,
+    Optional['DPoPMode']                                   => String,
+    Optional['FilterClaimsExpr']                           => String,
+    Optional['LogoutRequestParams']                        => Pattern[/^[A-Za-z0-9\-\._%]+=[A-Za-z0-9\-\._%]+(&[A-Za-z0-9\-\._%]+=[A-Za-z0-9\-\._%]+)*$/],
+    Optional['LogoutXFrameOptions']                        => String,
+    Optional['MetricsData']                                => String,
+    Optional['MetricsPublish']                             => String,
+    Optional['PassAccessToken']                            => Enum['On', 'Off'],
```

Comment:
We have the `Apache::OnOff` data type, which should be preferred (There are a bunch others in this file, maybe we can fix them too in another commit).